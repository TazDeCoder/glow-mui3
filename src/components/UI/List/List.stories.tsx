import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { ListItem } from '../ListItem';

import { List } from './List';

export default {
  title: 'Components/UI/List',
  component: List,
} as ComponentMeta<typeof List>;

export const Default: ComponentStory<typeof List> = (args) => (
  <List width={320} {...args}>
    <ListItem>Headline</ListItem>
    <ListItem>Headline</ListItem>
    <ListItem>Headline</ListItem>
  </List>
);
