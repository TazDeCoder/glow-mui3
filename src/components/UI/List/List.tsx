/** @jsxRuntime classic */
/** @jsx jsx */

import { forwardRef } from 'react';
import { jsx } from '@theme-ui/core';

import { Flex as _Flex, FlexProps, FlexOwnProps } from '../../Utils';
import type { Assign, ForwardRef } from '../../types';

const Flex = _Flex as React.ForwardRefExoticComponent<
FlexProps & React.RefAttributes<HTMLUListElement>
>;

export interface ListProps
  extends Assign<React.ComponentPropsWithRef<'ul'>, FlexOwnProps> {}

export const List: ForwardRef<HTMLUListElement, ListProps> = forwardRef(
  ({ children, sx, ...props }, ref) => (
    <Flex
      ref={ref}
      as="ul"
      flexDirection="column"
      py={2}
      sx={sx}
      {...(props as FlexProps)}
    >
      {children}
    </Flex>
  ),
);
