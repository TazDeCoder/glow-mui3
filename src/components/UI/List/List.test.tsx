import { render } from '@testing-library/react';

import { List } from './List';

describe('<List />', () => {
  describe('Snapshots', () => {
    it('renders correctly', () => {
      const { container } = render(
        <List>
          <li>Foo</li>
          <li>Bar</li>
          <li>Baz</li>
        </List>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
