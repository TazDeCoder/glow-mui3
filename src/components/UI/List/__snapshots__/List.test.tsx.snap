// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Snapshots renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <ul
    class="emotion-0"
  >
    <li>
      Foo
    </li>
    <li>
      Bar
    </li>
    <li>
      Baz
    </li>
  </ul>
</div>
`;
